File: game-server/src/main/java/work/fking/pangya/game/packet/handler/RareShopOpenPacketHandler.java
package work.fking.pangya.game.packet.handler;

import io.netty.channel.Channel;
import work.fking.pangya.game.packet.inbound.RareShopOpenPacket;
import work.fking.pangya.game.packet.outbound.RareShopOpenResponsePacket;
import work.fking.pangya.networking.protocol.InboundPacketHandler;

public class RareShopOpenPacketHandler implements InboundPacketHandler<RareShopOpenPacket> {

    @Override
    public void handle(Channel channel, RareShopOpenPacket packet) {
        channel.writeAndFlush(new RareShopOpenResponsePacket());
    }
}

File: game-server/src/main/kotlin/work/fking/pangya/game/packet/handler/RareShopOpenPacketHandler.kt
package work.fking.pangya.game.packet.handler

import io.netty.buffer.ByteBuf
import work.fking.pangya.game.GameServer
import work.fking.pangya.game.net.ClientPacketHandler
import work.fking.pangya.game.packet.outbound.RareShopOpenResponsePacket
import work.fking.pangya.game.player.Player

class RareShopOpenPacketHandler : ClientPacketHandler {
    override fun handle(server: GameServer, player: Player, packet: ByteBuf) {
        player.channel().writeAndFlush(RareShopOpenResponsePacket())
    }
}
